.sandbox/provision/lifecycle-asg/cleanup.sh

```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+source ./common.sh
+
+# cleanup leaked instances
+echo "Checking and cleaning up the leaked instances (if any)"
+INSTANCE_IDS="$(aws ec2 describe-instances --filters "Name=instance-state-name,Values=running" "Name=tag:$SANDBOX_ASG_TAG,Values=true" --query "Reservations[].Instances[?!(Tags[?Key=='aws:autoscaling:groupName'] || Tags[?Key==$SANBODX_ID_TAG])].InstanceId[]" | jq -cMr '.[]')" 
+[[ -z $INSTANCE_IDS ]] || aws ec2 terminate-instances --instance-ids "${INSTANCE_IDS[@]}"
```

Comment:
I'm afraid this won't work. the `${INSTANCE_IDS[@]}` will be only expanded correctly if `INSTANCE_IDS` is an array, but `INSTANCE_IDS="$(...)"` only makes it a space-separated string.

Also `-z $INSTANCE_IDS` should need quotes around `$INSTANCE_IDS`, otherwise if there are more than one token in `INSTANCE_IDS` it will result in syntax error.

Reply:
I think it works. `jq -cMr '.[]'` makes it a bash array. 

I tried it and `${INSTANCE_IDS[@]}` expanded.